monitoring/defender/autotask/on_module_factory_events.js

```diff
@@ -0,0 +1,1222 @@
+  const templateKeccak = utils.keccak256(templateBytes);
+  if (logs) {
+    const newRealityModuleProxies = logs
+      .filter((log) => log.topics !== null)
+      .filter((log) => log.topics[0] === keccak)
+      .map((log) => {
+        const proxy = utils.defaultAbiCoder.decode(
+          ["address"],
+          log.topics[1]
+        )[0];
+        const mastercopy = utils.defaultAbiCoder.decode(
+          ["address"],
+          log.topics[2]
+        )[0];
+        return {
+          proxy: ethers.utils.getAddress(proxy),
+          mastercopy: ethers.utils.getAddress(mastercopy),
+        };
+      })
+      // in tx it can be more than one proxy creation. We need to find only the ones that is the reality module
+      .filter(({ mastercopy }) => mastercopy != REALITY_MODULE_MASTERCOPY)
+      .map(({ proxy }) => proxy);
+
+    const newRealityModuleTemplate = logs
+      .filter((log) => log.topics !== null)
+      .filter((log) => log.topics[0] === templateKeccak);
+
+    const templateQuestionText = utils.defaultAbiCoder.decode(
+      ["string"],
+      newRealityModuleTemplate[0].data
+    );
+
+    return {
+      proxyAddresses: newRealityModuleProxies, // we must return all the proxies created in the tx (can be more then one)
+      templateQuestionText: templateQuestionText[0],
+    };
+  }
+};
+
+/**
+ *
+ * @param {number} questionTimeout
+ * @param {number} questionCooldown
+ * @param {number} bond
+ * @param {boolean} isController
+ * @param {boolean} isOwner
+ * @param {string} ensName
+ * @param {number} chainId
+ * @param {string} txHash
+ * @returns
+ */
+const generateDiscordParams = (
+  questionTimeout,
+  questionCooldown,
+  bond,
+  isController,
+  isOwner,
+  ensName,
+  chainId,
+  txHash
+) => {
+  const minBond = ethers.utils.parseUnits(MIN_BOND.toString(), 18);
+  const params = DISCORD_PARAMS;
+  let invalidTimeout = false;
+  let invalidCooldown = false;
+  let invalidBond = false;
+  let sendNotification = false;
+  if (questionTimeout < MIN_TIMEOUT) {
+    invalidTimeout = true;
+  }
+  if (questionCooldown < MIN_COOLDOWN) {
+    invalidCooldown = true;
+  }
+  if (bond.toString() < minBond.toString()) {
+    invalidBond = true;
+  }
+  console.log("invalidTimeout", invalidTimeout);
+  console.log("invalidCooldown", invalidCooldown);
+  console.log("invalidBond", invalidBond);
+  console.log("isOwner", isOwner);
+  console.log("isController", isController);
+  if (invalidTimeout) {
+    invalidInputs = `Timeout (Min expected -  ${MIN_TIMEOUT}) = ${questionTimeout}\n`;
+  }
+  if (invalidCooldown) {
+    invalidInputs = `${invalidInputs}Cooldown (Min expected -  ${MIN_COOLDOWN}) = ${questionCooldown}\n`;
+  }
+  if (invalidBond) {
+    invalidInputs = `${invalidInputs}Minimum Bond (Min expected - ${minBond.toString()}) = ${bond.toString()}\n`;
+  }
+  if (!isController) {
+    invalidInputs = `${invalidInputs}The ENS (${ensName}) controller is not the Avatar.\n`;
```

Comment:
```suggestion
    invalidInputs = `${invalidInputs}The Avatar is not the controller of the ENS name (${ensName}) mentioned in the template.\n`;
```

To clarify where we get the ENS from since it's not a trusted source.